statsforecast/arima.py

```diff
@@ -2085,4 +2086,284 @@ def try_params(p, d, q, P, D, Q, constant, k, bestfit):
+        self.start_Q=start_Q
+        self.stationary=stationary
+        self.seasonal=seasonal
+        self.ic=ic
+        self.stepwise=stepwise
+        self.nmodels=nmodels
+        self.trace=trace
+        self.approximation=approximation
+        self.method=method
+        self.truncate=truncate
+        self.test=test
+        self.test_kwargs=test_kwargs
+        self.seasonal_test=seasonal_test
+        self.seasonal_test_kwargs=seasonal_test_kwargs
+        self.allowdrift=allowdrift
+        self.allowmean=allowmean
+        self.blambda=blambda
+        self.biasadj=biasadj
+        self.parallel=parallel
+        self.num_cores=num_cores
+        self.period=period
+
+    def fit(self, y: np.ndarray, X: Optional[np.ndarray] = None):
+        """Fit the AutoARIMA estimator
+        Fit an AutoARIMA to a time series (numpy array) `y`
+        and optionally exogenous variables (numpy array) `X`.
+
+        Parameters
+        ----------
+        y: array-like of shape (n,)
+            One-dimensional numpy array of floats without `np.nan` or `np.inf`
+            values.
+        X: array-like of shape (n, n_x) optional (default=None)
+            An optional 2-d numpy array of exogenous variables (float).
+        """
+        self.model_ = auto_arima_f(
+            x=y,
+            d=self.d,
+            D=self.D,
+            max_p=self.max_p,
+            max_q=self.max_q,
+            max_P=self.max_P,
+            max_Q=self.max_Q,
+            max_order=self.max_order,
+            max_d=self.max_d,
+            max_D=self.max_D,
+            start_p=self.start_p,
+            start_q=self.start_q,
+            start_P=self.start_P,
+            start_Q=self.start_Q,
+            stationary=self.stationary,
+            seasonal=self.seasonal,
+            ic=self.ic,
+            stepwise=self.stepwise,
+            nmodels=self.nmodels,
+            trace=self.trace,
+            approximation=self.approximation,
+            method=self.method,
+            truncate=self.truncate,
+            xreg=X,
+            test=self.test,
+            test_kwargs=self.test_kwargs,
+            seasonal_test=self.seasonal_test,
+            seasonal_test_kwargs=self.seasonal_test_kwargs,
+            allowdrift=self.allowdrift,
+            allowmean=self.allowmean,
+            blambda=self.blambda,
+            biasadj=self.biasadj,
+            parallel=self.parallel,
+            num_cores=self.num_cores,
+            period=self.period
+        )
+
+        return self
+
+    def predict(self, h: int, X: Optional[np.ndarray] = None,
+                level: Optional[int] = None):
+        """Forecast future values using a fitted AutoArima.
+
+        Parameters
+        ----------
+        h: int
+            Number of periods for forecasting.
+        X: array-like of shape (n, n_x) optional (default=None)
+            Future exogenous variables.
+        level: int
+            Confidence level for prediction intervals.
```

Comment:
Maybe this can be an optional tuple instead to get several levels at once

Reply:
Nice! I think the second option is better, I'll include it :)